README
NAME
    Module::CPANTS::Anaylse - tests a dist for Kwalitee

DESCRIPTION
    This module is part of the beta CPANTS project.
    http://cpants.perl.org
    
AUTHOR
    Thomas Klausner, <domm@cpan.org>

COPYRIGHT

    Copyright (C) 2003-2009  Thomas Klausner
    Some files are Copyright (C) 2006-2008  Gabor Szabo
    Some files are Copyright (C) 2006  Steffen Müller

    The module distributions in the t/eg/ directory are copyright by
    their respective authors.

LICENSE
    This code is distributed under the same license as Perl.



Changes
#-----------------------------------------------------------------
# Changes for Module::CPANTS::Analyse
#-----------------------------------------------------------------

0.85    2009-07-12, Alexandr Ciornii
    - Fix fails on non-Win32

0.84    2009-07-08, Alexandr Ciornii
    - require Test::Warn 0.11
    - add Test::More to deps
    - add LWP::Simple to deps
    - fix calc_test_yaml_meta.t
    - better META.yml
    - list of changes for 0.83 (domm)


0.83    2009-06-10
    - ignore no_index when searching for examples (xdaveg)
    - skip the debian metrics and downloading the debian file when in LINT
      (szabgab)
    - has_test_pod and has_test_pod_coverage are now optional metrics (domm)


0.82	2008-07-08 10:23:30
    - update META.yml spec to 1.4
    - fixed experimental metrics
    - use Moose now counts for using strict
    - lots of commits from Gabor
    - honor META.yml key 'no_index'. Fixes RT#32777 and an IRC
      complaint by BinGOs


0.81   Sun Apr 13 16:45:29 CEST 2008
    - fixed doc bug in SYNOPSIS (reported by ADAMOWSKI as RT#34773)
    - made FindModules smarter (domm):
      If available, use META.yml 'provides'. If not, look for 'package'
      statements in files.
      It now also finds stuff like Tk::Widget and App::Ack
    - upgrade to Software::License 0.003 (Gabor)


0.80   Sat Apr 12 10:53:40 CEST 2008
    domm:
    - add metayml_has_provides metric
    Gabor Szabo:
    - simplify the way the VERSION number is looked up, keep the full version line
    - increase the no_large_files limit to 200_000
    - some unit test fixes thanks to Tux
    - some of {error} fields are now ARRAY refs instead of strings
    - more detailed unit tests
    - add LICENCE to the list of special files (not only LICENSE)
    - collect the license information of each file using Software::LicenseUtils
    - add has_separate_license_file metric
    - add has_license_in_source_file metric


0.79  Tue Apr  8 01:25:12 CEST 2008
    * released at the Oslo QA Hackathon 2008
    Gabor Szabo:
    - replace File::Find by File::Find::Rule
    - add no_large_files metrics
    - add fits_fedora_license metric that was inside the easily_repackageable_by_fedora metric
    - add is_experimental flag to some of the metrics
    - add aggregating key to metric to list the other metrics being aggregated
    - unhide some errors during testing
    - renamed some error fields
    - add metayml_declares_perl_version metric


0.78    2008-04-06
    * released at the Oslo QA Hackathon 2008
    - very hacky workaround for problems with Meta.YML, version 
      objects, etc
    - moved has_version_in_each_file into MCK::Version & fixed a bug
      in it 
    Gabor Szabo:
    - add fits_fedora_license metric that was inside the easily_repackageable_by_fedora metric
    - add is_experimental and is_aggregate flags to some of the metrics
    - unhide some errors during testing
    - add no_generated_files metric
    - add easily_repackageagble_by_fedora, easily_repackagable_by_debian
      and easily_repackageagble metric
    - add has_version_in_each_file metric
    - add has_tests_in_t_dir metric
    - add no_stdin_for_prompting metric checking for STDIN in Makefile.PL and Build.PL


0.77	2008-01-29
    - made 'prereq_matches_use' and 'build_prereq_matches_use' optional metrics
    - changed data structure: all errors are now in a hashref called 'error'


0.76    2007-11-14
    - added new metrics 'prereq_matches_use' and
      'build_prereq_matches_use'
    - added 'needs_db' to metrics and removed hardcoded workaround for
      is_prereq
    - removed workaround in MCK::MetaYML (Barbie fixed it the problem
      in Test::YAML::Meta)
    - reworded metayml_is_parsable remedy


0.75    2007-10-29
    - added optional metric use_warnings
    - added directory 'xt' as a potential location for test files:w
    - finally applied patch provided by Adriano Ferreira to fix some
      issues in MCK::Manifest. See RT #28982


0.74    2007-10-24
    - removed metric metayml_conforms_spec_1_0
    - fixed bug in MCK::MetaYML::check_spec_conformance, which caused 
      wrong results in metayml_conforms_spec*
    - switched to YAML::Syck
    - work around Pod::Simple::Checker returning strange data


0.73    2007-09-12
    - added version of dist to dump file name
    - fixed bug in MCK::NeedsCompiler (RT #28134 plus all reported merged
      into that one)


0.72    2007-06-30
    - added docs to cpants_lint.pl 
    - cpants_lint.pl can now dump metadata to a file
    - added YAML output to cpants_lint.pl 
    Gabor Szabo:
    - added MCK::NeedsCompiler
      MCK::NeedsCompiler does not contain any kwalitee_indicator, but
      adds much needed metadata.
    Barbie:
    - MCA assumes Unix paths after munging with File::Spec, thus breaking 
      portability at every turn in a non-Unix-like environment
      ... bad Thomas no biscuit!
    Barbie:
    - Refined META.yml metrics using Test::YAML::Meta and brought 'current
      spec' up to date with version 1.3 :)


0.71    2007-04-23
    - moved svn repo to http://code.google.com/p/cpants/
    - has_license now also checks for LICENCE (as suggested by David
      Cantrell)
    - fixed RT #26535 reported by MTHURN
    - has_examples now also checks for p(m|od) file (as suggested by
      JUERD)
    - fixed RT #24228 reported by THALJEF - t/99_critic.t now only runs
      if $ENV{AUTHOR_TEST} is set
    - resolved RT #25198 reported by bdfoy using his patch
    - FindModules: do not check META.yml for provides because META.yml
      might list semi-garbage (eg according to META.yml, Siesta provides
      Apache::Constants, which it does not.)
    - skip very large test-files and only check *.t files (not everything
      in t/) as suggested by CASIANO
    - cpants_lint.pl:

Build.PL
use strict;
use Module::Build;

my $build = Module::Build->new(
    module_name => 'Module::CPANTS::Analyse',
    license     => 'perl',
    requires    => {
        'Class::Accessor'           => '0.19',
        'Archive::Any'              => '0.06',
        'Archive::Tar'              => '1.30',
        'CPAN::DistnameInfo'        => '0.06',
        'Pod::Simple::Checker'      => '2.02',
        'YAML::Syck'                => '0.95',
        'Module::ExtractUse'        => '0.18',
        'Module::Pluggable'         => '2.96',
        'IO::Capture'               => '0.05',
        'Array::Diff'               => '0.04',
        'Test::YAML::Meta::Version' => '0.11',
        'version'                   => '0.73',
        'List::MoreUtils'           => '0',
        'File::Find::Rule'          => '0',
        'Readonly'                  => '0',
        'Software::License'         => '0.003',
        'File::Slurp'               => '0',
        'Text::CSV_XS'              => '0.45',
        'LWP::Simple'               => 0,
        'perl'                      => 5.006,
    },
    build_requires => {
        'Test::More'                => '0',
        'Test::Deep'                => '0',
        'Test::Warn'                => '0.11', #less problems with new Sub::Uplevel
        'Test::NoWarnings'          => '0',
    },
    meta_merge => {
        resources => {
            repository => 'http://github.com/domm/CPANTS/tree',
        },
        keywords => [
            'CPANTS','kwalitee',
        ],
    },
    script_files=>[qw(bin/cpants_lint.pl)],
    create_makefile_pl=>'traditional',
);


$build->create_build_script;

